com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -17,23 +60,28 @@ var FileLoader = {
                 var onprogress = this.onprogress;
                 var onload = this.onload;
                 var onerror = this.onerror;
+                var onretry = this.onretry;
 
                 var xhr = new XMLHttpRequest();
+                xhr._loadedSize = 0;
                 xhr.open(method, url, true);
                 xhr.responseType = responseType;
-                xhr.onprogress = function(e) {
-                    if (onprogress) onprogress(xhr, e);
+                xhr.onprogress = function(event) {
+                    if (onprogress) onprogress(xhr, event, xhr._loadedSize);
+                    xhr._loadedSize = event.loaded;
                 };
-                xhr.onerror = function(e) {
+                xhr.onerror = function(event) {
                     if (currentAttempt == FileLoader.options.retryCount) {
-                        if (onerror) onerror(xhr, e);
+                        if (onerror) onerror(xhr, event);
                         return;
                     }
-                    currentAttempt = currentAttempt + 1;
+                    if (onretry) onretry(xhr, event, xhr._loadedSize, currentAttempt);
```

Comment:
Add `onretry` callback. It uses to decrease loaded amount of bytes and update progress bar with correct values.